### DISCUSSION TITLE
Which json elements experts for auth

### DISCUSSION THREAD

**Discussion by jasonmellone** [KmiQ/coinbase-advanced-python#41] [Q&A] [answered]
Would anyone mind calling out the specific keys required to be sent for the auth workflow when initializing the client? I've used various clients in the past and had managed when things were cbpro. Maybe something is wrong with my api key but want to be sure I eliminate this as a potential error on my end. 

Right now getting an unauthorized error w what I'm passing. 

Thank you!

**Comment by KmiQ** [comment]
Hey Jason, thank you for your question.

I'm working on the next release ([PR here](https://github.com/KmiQ/coinbase-advanced-python/pulls)) and I'm adding factory methods to initialize the client for the different auth schemas explicitly. Do you think that will be enough or do you have something else in mind?

**Reply by KmiQ** [reply, marked as answer]
You can see the factory I'm adding: https://github.com/KmiQ/coinbase-advanced-python/pull/40/files#diff-dda42262ec41b8f8f55890f62aa38a0c5cfeee673b18f90df167d20175076cfeR54-R71

**Reply by jasonmellone** [reply]
Thanks this is resolved for me.

For anyone else:
```

    config = json.load(open('~/cloud_api_key.json','r'))
    # Creating the client using Clould API Keys.
    client = CoinbaseAdvancedTradeAPIClient.from_cloud_api_keys(config['name'], config['privateKey'])

```

Where `coud_api_key.json` is generated from the coinbase website.
